app/email.go

```diff
@@ -513,6 +513,53 @@ func (a *App) SendDeactivateAccountEmail(email string, locale, siteURL string) *
 	return nil
 }
 
+func (a *App) SendWarnMetricAckEmail(warnMetricId string, email string) *model.AppError {
+	if len(*a.Config().EmailSettings.SMTPServer) == 0 {
+		return model.NewAppError("SendWarnMetricAckEmail", "api.email.send_warn_metric_ack.missing_server.app_error", nil, utils.T("api.context.invalid_param.app_error", map[string]interface{}{"Name": "SMTPServer"}), http.StatusInternalServerError)
+	}
+
+	var subject string
+	var body string
+
+	switch warnMetricId {
+	case model.SYSTEM_WARN_METRIC_NUMBER_OF_ACTIVE_USERS:
+		subject = "Acknowledge of User Limit"
+		body = "This is a receipt of acknowledgement for the number of active users exceeding the limit for the following site.\r\n"
+
+		body += fmt.Sprintf("Contact Email: %s\r\n", email)
+		body += fmt.Sprintf("DiagnosticId: %s\r\n", a.DiagnosticId())
+		body += fmt.Sprintf("SiteURL: %s\r\n", a.GetSiteURL())
+
+		if a.License() != nil {
+			body += fmt.Sprintf("License Id: %s\r\n", a.License().Id)
+		}
+
+		//same definition as the active users count metric displayed in the SystemConsole Analytics section
+		registeredUsersCount, err := a.Srv().Store.User().Count(model.UserCountOptions{})
+		if err != nil {
+			mlog.Error("Error retrieving the number of registered users", mlog.Err(err))
+		} else {
+			body += fmt.Sprintf("Registered Users: %v\r\n", registeredUsersCount)
+		}
+		body += "If you have any additional inquiries, please contact support@mattermost.com"
+	default:
+		return model.NewAppError("SendWarnMetricAckEmail", "api.email.send_warn_metric_ack.invalid_warn_metric.app_error", nil, "", http.StatusInternalServerError)
+	}
+
+	if err := mailservice.SendMailUsingConfig(model.MM_ACKNOWLEDGE_ADDRESS, subject, body, a.Config(), false, email); err != nil {
+		mlog.Error("Error while sending email", mlog.String("email", model.MM_ACKNOWLEDGE_ADDRESS), mlog.Err(err))
+		return model.NewAppError("SendWarnMetricAckEmail", "api.email.send_warn_metric_ack.failure.app_error", map[string]interface{}{"Error": err.Error()}, "", http.StatusInternalServerError)
+	}
+
+	mlog.Info("Disable the monitoring of warn metric", mlog.String("metric", warnMetricId))
+	err := a.SetWarnMetricStatus(warnMetricId)
```

Comment:
because this is here, I think you are unable to ack the warning unless you have the email configured. I think the idea was, when you "ACK" the warning, we accept the ACK optimistically and then try to send the email.

Reply:
The behavior, to my understanding is:
- click the "Acknowledgement" button in the UX, this tries to send the email and store the ack in the DB
- if sending the email fails (because it is not configured on the server or some other reasons), we are not storing the ack and instead we are showing the user a "MailTo" link in the modal. If they click it, we send the ack and open the mailto link in the browser. There is a functionality gap/hole right now in that we do not check if the user sent the mailto mail in the browser (we are instead acking the moment they click on the mailto link). I updated the logic to handle the forceAck flag.